File: languages/php/Bloock/Entity/Identity/Identity.php
<?php

namespace Bloock\Entity\Identity;

class Identity
{
    private string $mnemonic;
    private string $key;
    private string $privateKey;

    /**
     * @param string $mnemonic
     * @param string $key
     * @param string $privateKey
     */
    public function __construct(string $mnemonic, string $key, string $privateKey)
    {
        $this->mnemonic = $mnemonic;
        $this->key = $key;
        $this->privateKey = $privateKey;
    }

    public static function fromProto(\Bloock\Identity $res): Identity
    {
        return new Identity($res->getMnemonic(), $res->getKey(), $res->getPrivateKey());
    }

    /**
     * @return string
     */
    public function getMnemonic(): string
    {
        return $this->mnemonic;
    }

    /**
     * @return string
     */
    public function getKey(): string
    {
        return $this->key;
    }

    /**
     * @return string
     */
    public function getPrivateKey(): string
    {
        return $this->privateKey;
    }

    public function toProto(): \Bloock\Identity
    {
        $p = new \Bloock\Identity();
        $p->setMnemonic($this->mnemonic);
        $p->setKey($this->key);
        $p->setPrivateKey($this->privateKey);
        return $p;
    }
}

File: languages/python/bloock/entity/identity/identity.py
from __future__ import annotations

import bloock._bridge.proto.identity_entities_pb2 as proto


class Identity:
    def __init__(self, mnemonic: str, key: str, private_key: str) -> None:
        self.mnemonic = mnemonic
        self.key = key
        self.private_key = private_key

    @staticmethod
    def from_proto(c: proto.Identity) -> Identity:
        return Identity(
            mnemonic=c.mnemonic,
            key=c.key,
            private_key=c.private_key
        )

    def to_proto(self) -> proto.Identity:
        return proto.Identity(
            mnemonic=self.mnemonic,
            key=self.key,
            private_key=self.private_key,
        )
